### DISCUSSION TITLE
Push reactive changesets into PersistedCollection

### DISCUSSION THREAD

**Discussion by jamesgibson14** [maxnowack/signaldb#765] [Ideas]
Currently `onChange` just triggers a `this.persistenceAdapter.load()`. Could we pass a changeSet to onChange -> loadPersistentData so that it could just load the partial changes instead of doing a full load/pull? Something like:
```javascript
async register(onChange) {
      websock.onMessage({items, changes} => {
        void onChange({items, changes})
      }
    },
```
if `loadPersistentData` receives a changeSet then it just uses that instead of calling `load`

**Comment by maxnowack** [comment]
I think it's a good idea. This makes the development of new persistence adapters more straight forward 🙂
